### PR TITLE
WT-12709 Fix timestamp of reinstantiated deletes being overwritten at commit

### PR DESCRIPTION
**Pull request by quchenhao**

### CODE REVIEW

**Comment by etienneptl** [comment]
What is the fix in this PR compared to the previous one and can I have a link to the patch build?

**Comment by quchenhao** [comment]
@etienneptl [aba212e](https://github.com/wiredtiger/wiredtiger/pull/10411/commits/aba212ed51fce7b6b298078c1448d9a7a9775eee)

**Comment by etienneptl** [comment]
> @etienneptl [c36606b](https://github.com/wiredtiger/wiredtiger/pull/10411/commits/c36606bfdf9e306772cd644d8c06d416ca65ffaa) [aba212e](https://github.com/wiredtiger/wiredtiger/pull/10411/commits/aba212ed51fce7b6b298078c1448d9a7a9775eee)

Sorry I missed the commits, thank you!

**Review by etienneptl — APPROVED** [review]
LGTM!

File: test/suite/test_truncate24.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        self.session.commit_transaction('commit_timestamp=' + self.timestamp_str(20))
+
+        # Check stats to make sure we fast-deleted at least one page.
+        stat_cursor = self.session.open_cursor('statistics:', None, None)
+        fastdelete_pages = stat_cursor[wiredtiger.stat.conn.rec_page_delete_fast][2]
+        self.assertGreater(fastdelete_pages, 0)
+
+        # Verify we don't see the data at timestamp 10.
+        for i in range(1, 100000):
+            self.session.begin_transaction("read_timestamp=" + self.timestamp_str(10))
```
Comment (etienneptl) [inline test/suite/test_truncate24.py:83]: Do you need to do this within the `for` loop?
Same for the second test.

File: test/suite/test_truncate24.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+from wtdataset import SimpleDataSet
+from wtscenario import make_scenarios
+
+class test_truncate24(wttest.WiredTigerTestCase):
+    key_format_values = [
+        ('row', dict(key_format='i', value_format='i')),
+        ('var', dict(key_format='r', value_format='i')),
+    ]
+
+    scenarios = make_scenarios(key_format_values)
```
Comment (etienneptl) [inline test/suite/test_truncate24.py:43]: We can use `make_scenarios` to have one scenario with and w/o ts.

File: test/suite/test_truncate24.py (hunk 1 above)
Comment (quchenhao) [reply test/suite/test_truncate24.py:83]: I did this because I don't want to have a long running transaction. I can try have the transaction outside.

Comment (etienneptl) [reply test/suite/test_truncate24.py:83]: I think it's good as it is, let's not risk the long-running txn, there are no benefits.

**Review by etienneptl — APPROVED** [review]

**Pull request merged**